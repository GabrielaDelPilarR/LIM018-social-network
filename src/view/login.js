import { loginGoogle } from '../firebase/auth.js';

const login = {
  template: () => {
    const login = `
      <section class="firstView">
          <div class="containerIndex">
            <img src="./img/cubiertos.png" alt="Cubiertos">
            <h1> HELP TASTER </h1>
            <h2> ¿Que deseas probar hoy? </h2>
            <span> Te ayudamos a encontrar los mejores lugares para disfrutar tus comidas!!</span>
            <div class="buttonIndex">
              <a href="#/home" class="button" id="googleLogin">
                <img src="./img/Google.png">
                Inicia con Google</a>
              <a href="#/facebook" class="button">
                <img src="./img/Facebook.png">
                Inicia con Facebook</a>
              <a href="#/loginEmail" class = "button">
                <img src="./img/E-mail.png">
                Inicia con Email</a>
            </div>
            <span>¿No tienes una cuenta ?</span>
            <a href="#/register"> Registrate aqui</a>
          </div>  
      </section>`;

    const divLogin = document.createElement('div');
    divLogin.classList.add('login');
    divLogin.innerHTML = login;
    divLogin.querySelector('#googleLogin').addEventListener('click', loginGoogle);
    return divLogin;
  },

  // init: ()=> {
  // const btnEmail = document.querySelector('#btn-email')
  // btnEmail.addEventListener('click',() => {
  // changeView(window.location.hash);

  // })
  // }

};

export default login;
