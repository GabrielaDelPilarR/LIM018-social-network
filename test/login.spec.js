import login from '../src/view/login.js';
/* import logIn from '../src/firebase/auth.js'; */

jest.mock('../src/firebase/auth.js', () => ({
  loginGoogle: jest.fn(),
  logIn: jest.fn().mockImplementationOnce(() => Promise.resolve({
    displayName: 'Karla', email: 'karlavasquez817@gmail.com', uid: 'jdfhds4545248',
  })),
  sendEmailVerification: jest.fn(),
}));

const renderLogin = () => {
  window.location.href = '';
  document.body.innerHTML = '';
  document.body.appendChild(login.template());
  login.init();
};

describe('login', () => {
  it('should be a function', () => {
    expect(typeof login.template).toBe('function');
    expect(typeof login.init).toBe('function');
  });

  it('should check template', () => {
    expect(login.template()).toMatchSnapshot();
  });

  it('should render template', () => {
    renderLogin();
    const elem = document.querySelector('.form-email');
    expect(elem instanceof HTMLElement).toBeTruthy();
  });

  it('should close modal', () => {
    renderLogin();
    const btnModal = document.querySelector('.btn-modal');
    const modal = document.querySelector('.modal-container');
    btnModal.click();
    modal.classList.remove('show-modal');
    expect(modal instanceof HTMLElement).toBeTruthy();
  });

  it('should render template login', () => {
    renderLogin();

    const formEmail = document.querySelector('.form-email');
    formEmail.submit();
    const messageError = document.querySelector('.message-error');
    expect(messageError.classList.contains('show-message-error')).toBeTruthy();

    const inputEmail = document.querySelector('#inputEmail');
    const inputPassword = document.querySelector('#inputPassword');
    inputEmail.value = 'karlavasquez817@gmail.com';
    inputPassword.value = '1234567';
    formEmail.submit();
    expect(messageError.classList.contains('show-message-error')).toBeFalsy();
  });

  it('No deberia abrirse el modal si el email esta verificado', () => {
    renderLogin();

    const modalContainer = document.querySelector('.modal-container');
    const btnModal = document.querySelector('.btn-log-email');
    btnModal.click();
    expect(modalContainer.classList.contains('show-modal')).toBeFalsy();
  });
});
