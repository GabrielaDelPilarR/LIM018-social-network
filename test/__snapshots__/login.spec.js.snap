// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login should check template 1`] = `
<div
  class="login"
>
  
      
  <section
    class="first-view"
  >
    
        
    <div
      class="container-index"
    >
      
          
      <img
        alt="Cubiertos"
        src="./img/cubiertos.png"
      />
      
          
      <h1>
         HELP TASTER 
      </h1>
      
          
      <h2>
         ¿Que deseas probar hoy? 
      </h2>
      
          
      <span>
         Te ayudamos a encontrar los mejores lugares para disfrutar tus comidas!!
      </span>
      
          
      <form
        class="form-email"
      >
        
            
        <input
          class="input-register"
          id="inputEmail"
          name="email"
          placeholder="E-mail"
          type="email"
        />
        
            
        <p
          class="condicion loginError"
          id="error-usuario"
        />
        
            
        <input
          class="input-register"
          id="inputPassword"
          name="password"
          placeholder="Contraseña"
          type="password"
        />
        
            
        <p
          class="condicion loginError"
          id="error-contraseña"
        />
        
            
        <button
          class="btn-log-email"
          type="submit"
        >
           Iniciar sesión 
        </button>
        
          
      </form>
      
          
      <span
        class="message-error"
      >
        Llena los campos
      </span>
      
          
      <div
        class="separator"
      >
        <hr
          class="hr"
        />
        O
        <hr
          class="hr"
        />
      </div>
      
          
      <div
        class="button-index"
      >
        
            
        <button
          class="button"
          id="googleLogin"
        >
          
              
          <img
            src="./img/Google.png"
          />
          
              Inicia con Google
            
        </button>
        
          
      </div>
      
          
      <span>
        ¿No tienes una cuenta ?
      </span>
      
          
      <a
        href="#/register"
      >
         Registrate aqui
      </a>
      
        
    </div>
      
        
    <div
      class="modal-container"
    >
      
          
      <div
        class="modal no-verified-email"
      >
        
            
        <img
          src="./img/verificacion.webp"
        />
         
            
        <span>
           No se verifico, revisa su correo
        </span>
        
            
        <button
          class="btn-modal"
        >
           Aceptar 
        </button>
        
          
      </div>
      
        
    </div>
    
      
  </section>
</div>
`;
